lib/components/viewers/route-details.js

```diff
@@ -0,0 +1,203 @@
+
+    const headsigns =
+      patterns &&
+      Object.entries(patterns)
+        .map((pattern) => {
+          return {
+            geometryLength: pattern[1].geometry?.length,
+            headsign: extractHeadsignFromPattern(pattern[1]),
+            id: pattern[0]
+          }
+        })
+        // Remove duplicate headsigns. Using a reducer means that the first pattern
+        // with a specific headsign is the accepted one. TODO: is this good behavior?
+        .reduce((prev, cur) => {
+          const amended = prev
+          const alreadyExistingIndex = prev.findIndex(
+            (h) => h.headsign === cur.headsign
+          )
+          // If the item we're replacing has less geometry, replace it!
+          if (alreadyExistingIndex >= 0) {
+            // Only replace if new pattern has greater geometry
+            if (
+              amended[alreadyExistingIndex].geometryLength < cur.geometryLength
+            ) {
+              amended[alreadyExistingIndex] = cur
+            }
+          } else {
+            amended.push(cur)
+          }
+          return amended
+        }, [])
+        .sort((a, b) => {
+          // sort by number of vehicles on that pattern
+          const aVehicleCount = route.vehicles?.filter(
+            (vehicle) => vehicle.patternId === a.id
+          ).length
+          const bVehicleCount = route.vehicles?.filter(
+            (vehicle) => vehicle.patternId === b.id
+          ).length
+
+          // if both have the same count, sort by pattern geometry length
+          if (aVehicleCount === bVehicleCount) {
+            return b.geometryLength - a.geometryLength
+          }
+          return bVehicleCount - aVehicleCount
+        })
+
+    return (
+      <Container backgroundColor={routeColor} textColor={textColor}>
+        <RouteNameContainer className={className} textColor={textColor}>
+          <LogoLinkContainer>
+            {agency && <FormattedMessage
+              id='components.RouteDetails.operatedBy'
+              values={{
+                agencyName: agency.name
+              }}
+            />}
+            {url && (
+              <MoreDetailsLink color={textColor} href={url} target='_blank'>
+                <Icon type='link' />
+                <FormattedMessage id='components.RouteDetails.moreDetails' />
+              </MoreDetailsLink>
+            )}
+          </LogoLinkContainer>
+        </RouteNameContainer>
+        <PatternContainer color={routeColor} textColor={textColor}>
+          <h4><FormattedMessage id='components.RouteDetails.stopsTo' /></h4>
+          {headsigns &&
+            headsigns.map((h) => (
+              <Button
+                bsStyle='link'
+                className={h.id === pattern?.id ? 'active' : ''}
+                key={h.id}
+                onClick={() => this._headSignButtonClicked(h.id)}
+              >
+                {h.headsign}
+              </Button>
+            ))}
```

Comment:
Because the text in there can be long and there can be more that 2 patterns, I am starting to think that a `<select>` would be more appropriate, what do you think?

Reply:
I've implemented `<select>`, but think we should consider going back to buttons because it slows the user down, especially on mobile.

Reply:
in c6723870a8fc3a5bbb213786282e05ec6fa7d045